src/modules/ToDoList.js

```diff
@@ -0,0 +1,133 @@
+export default class ToDoList {
+  constructor(task, completed, index) {
+    this.task = task;
+    this.completed = completed;
+    this.index = index;
+  }
+}
+
+const section = document.querySelector('section');
+section.innerHTML = `
+  <div class="main-div">
+    <p id="header">Today's To Do <i class="fas fa-sync"></i></p>
+    <form class="form">
+      <input class="dataEntry" type="text" placeholder="Add to your list..." required></input>
+    </form>
+    <p id="clear">Clear all completed</p>
+  </div>
+`;
+
+// Create list
+const createList = () => {
+  const form = document.querySelector('.form');
+  const list = document.createElement('div');
+  list.className = 'input-div';
+  form.appendChild(list);
+  const checkboxes = document.createElement('input');
+  checkboxes.className = 'input';
+  checkboxes.type = 'checkbox';
+  const listText = document.createElement('p');
+  listText.className = 'listContent';
+  const threeDots = document.createElement('i');
+  threeDots.className = 'fas fa-ellipsis-v';
+  const trashIcon = document.createElement('i');
+  trashIcon.className = 'fas fa-trash-alt icon2';
+  list.append(checkboxes, listText, threeDots, trashIcon);
+  // Add event to checkboxes
+
+  checkboxes.addEventListener('click', () => {
+    threeDots.classList.toggle('remove-icon-active');
+    trashIcon.classList.toggle('icon2');
+    listText.classList.toggle('listContent-disable');
+    list.classList.toggle('newbackg');
+    const achieveStorage = JSON.parse(localStorage.getItem('list'));
+    const newEvent = [];
+    const newTask = document.querySelectorAll('.input-div');
+    for (let i = 0; i < achieveStorage.length; i += 1) {
+      if (newTask[i].classList.contains('newbackg')) {
+        achieveStorage[i].completed = true;
+      } else {
+        achieveStorage[i].completed = false;
+      }
+      newEvent.push(achieveStorage[i]);
+      localStorage.setItem('list', JSON.stringify(newEvent));
+    }
+  });
+
+  // Remove from list event
+  trashIcon.addEventListener('click', () => {
```

Comment:
- [ ] Nice work here, there seems to be an issue when deleting a task, it deletes multiple tasks when you select 1 and the last task is not been deleted, also has a requirement Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. find below a sample of how you can implement this.

```
trashIcon.addEventListener('click', () => {
    form.removeChild(list);
    const getFromLocalStorage = JSON.parse(localStorage.getItem('list'));
   let result;
    if(getFromLocalStorage) {
      result = getFromLocalStorage.filter((x, i) => x.task !== listText.textContent);
    }
    const reset = result.map((e, i) => {
      e.index = i + 1;
      return e;
    });
    localStorage.setItem('list', JSON.stringify(reset));
  });
```